lib/http_accept_language/auto_locale.rb

```diff
@@ -11,7 +11,9 @@ module AutoLocale
     private
 
     def set_locale
-      I18n.locale = http_accept_language.compatible_language_from(I18n.available_locales)
+      hal = http_accept_language
+      hal.header ||= I18n.default_locale.to_s
+      I18n.locale = hal.compatible_language_from(I18n.available_locales)
```

Comment:
@knu I'm not sure I understand. If none of the locales in `Accept-Language` are supported by the website, why _wouldn't_ you want to fall back to the default locale?

With your code, `I18n.locale` can still end up being set to `nil`, which, as your write, leads to a "nil is not a valid locale" error.